File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.java
import java.util.*;

class Solution {
    public int solution(int[] topping) {
        Set<Integer> older = new HashSet<>();
        Map<Integer, Integer> younger = new HashMap<>();
        
        int answer = 0;
        for (int t : topping) {
            younger.put(t, younger.getOrDefault(t, 0)+1);
        }
        
        for (int t : topping) {
            older.add(t);
            younger.put(t, younger.get(t) - 1);
            
            if (younger.get(t) == 0) younger.remove(t);
            
            if (younger.size() == older.size()) answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/lv2/132265. 롤케이크 자르기/롤케이크 자르기.py
from collections import Counter

def solution(topping):
    answer = 0
    
    older = set()
    younger = Counter(topping)
    
    for t in topping:
        older.add(t)
        younger[t] -= 1
        if younger[t] == 0:
            del younger[t]

        answer += len(younger) == len(older)
            
    return answer